static/app/views/alerts/rules/metric/ruleForm.tsx

```diff
@@ -1283,98 +1283,118 @@ class RuleFormContainer extends DeprecatedAsyncComponent<Props, State> {
-            isMigration && !triggersHaveChanged ? t('Looks good to me!') : t('Save Rule')
-          }
+        <OnDemandThresholdChecker
+          projectId={project.id}
+          isExtrapolatedChartData={!!isExtrapolatedChartData}
         >
-          <List symbol="colored-numeric">
-            <RuleConditionsForm
-              aggregate={aggregate}
-              alertType={alertType}
-              allowChangeEventTypes={
-                dataset === Dataset.ERRORS || alertType === 'custom_transactions'
-              }
-              comparisonDelta={comparisonDelta}
-              comparisonType={comparisonType}
-              dataset={dataset}
-              disableProjectSelector={disableProjectSelector}
-              disabled={formDisabled}
-              isEditing={Boolean(ruleId)}
-              isErrorMigration={showErrorMigrationWarning}
-              isExtrapolatedChartData={isExtrapolatedChartData}
-              isForLlmMetric={[
-                'sum(ai.total_tokens.used)',
-                'sum(ai.total_cost)',
-              ].includes(aggregate)}
-              isTransactionMigration={isMigration && !showErrorMigrationWarning}
-              onComparisonDeltaChange={value =>
-                this.handleFieldChange('comparisonDelta', value)
-              }
-              onFilterSearch={this.handleFilterUpdate}
-              onTimeWindowChange={value => this.handleFieldChange('timeWindow', value)}
-              organization={organization}
-              project={project}
-              router={router}
-              thresholdChart={wizardBuilderChart}
-              timeWindow={timeWindow}
-            />
-            <AlertListItem>{t('Set thresholds')}</AlertListItem>
-            {thresholdTypeForm(formDisabled)}
-            {showErrorMigrationWarning && (
-              <Alert type="warning" showIcon>
-                {tct(
-                  "We've added [code:is:unresolved] to your events filter; please make sure the current thresholds are still valid as this alert is now filtering out resolved and archived errors.",
-                  {
-                    code: <code />,
-                  }
-                )}
-              </Alert>
-            )}
-            {triggerForm(formDisabled)}
-            {ruleNameOwnerForm(formDisabled)}
-          </List>
-        </Form>
+          {({isOnDemandLimitReached}) => {
+            const submitDisabled =
+              formDisabled ||
+              !this.state.isQueryValid ||
+              (isExtrapolatedChartData &&
+                isOnDemandQueryString(this.state.query) &&
+                isOnDemandLimitReached);
+            return (
+              <Form
+                model={this.form}
+                apiMethod={ruleId ? 'PUT' : 'POST'}
+                apiEndpoint={`/organizations/${organization.slug}/alert-rules/${
+                  ruleId ? `${ruleId}/` : ''
+                }`}
+                submitDisabled={submitDisabled}
```

Comment:
It would be helpful to show a tooltip explaining why the button is disabled, but implementing this would require changes to the Form component, which is outside the scope of this PR.